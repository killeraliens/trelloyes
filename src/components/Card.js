import React from 'react';
import './Card.css';



class Card extends React.Component {
  render() {
    console.log(this.props)
    return (
      <div id={this.props.id} className='Card'>
        <h3>{this.props.title}</h3>
        <p>
          {this.props.content}
          <button>delete</button>
        </p>

      </div>
    );
  }
}

export default Card;
