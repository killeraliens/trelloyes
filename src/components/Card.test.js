import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Card from './Card';

describe('Card component', () => {

  it('renders Card component', ()=>{
    const div = document.createElement('div');
    ReactDOM.render(<Card />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders to the UI as expected', ()=> {
    const tree = renderer.create(<Card key='a' id='a' title='card title' content='some content' />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});

