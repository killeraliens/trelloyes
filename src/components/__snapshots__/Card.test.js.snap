// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders to the UI as expected 1`] = `
<div
  className="Card"
  id="a"
>
  <h3>
    card title
  </h3>
  <p>
    some content
    <button>
      delete
    </button>
  </p>
</div>
`;
